### DISCUSSION TITLE
Using colorMode in text / layer styles

### DISCUSSION THREAD

**Discussion by cysabi** [chakra-ui/chakra-ui#3450] [Q&A] [answered]
Hey! I am trying to create dark mode variants inside of my theme, so I do not need to specify the dark mode variant in each component I create.

I've chosen text & layer styles to be my preferred way of doing so. However, I can't figure out how to use dark mode inside of that within the theme.

Is there a better way, or am I doing something incorrectly?

**Comment by dodas** [comment, marked as answer]
This is duplicate of #2231.

You can use a workaround like this:
```tsx
const theme = {
  layerStyles: {
    yourStyle: {
      bg: "red.500",
      ".chakra-ui-dark &": {
        bg: "red.300",
      },
    },
  },
}
```

I would recommend using variants though. For existing Chakra UI components, you can either extend their variants, or create new ones. You can also style your own components via Chakra's theme using the `useStyleConfig` hook.
Read more about that here: https://chakra-ui.com/docs/theming/component-style

Does that help?

**Reply by cysabi** [reply]
hey, after reading the docs, I'm having trouble understanding the purpose of `useStyleConfig`.

My main issue with creating variants is that they are specific to a component. However, I want these styles to be able to apply to any kind of component.

**Reply by dodas** [reply]
Yes, for reusable styles applicable to any component, `layerStyle` is probably the way to go.
However, I think you should reconsider your approach. Personally I have never styled an application this way. You should try to create reusable components, and these components can be styled via theme, using the `useStyleConfig` hook.
In cases where you still decide for `layerStyles`, you can use the `".chakra-ui-dark &"` workaround for now.
